### ISSUE TITLE
Differences between Catboost.train in R and CatboostClassifier in Python.

### ISSUE THREAD

**Issue by alvaropc** [catboost/catboost#321]
Hi! 
I'm trying to compare the results I'm obtaining using Catboost.train in R and CatboostClassifier in Python.

I noticed that while running the predict function using class_type='Class' in both languages and using default parameters in order to try to code exactly the same, I get different values which result in a really different percentage of success while comparing the results to the real target. So I think there are some differences between Catboost.train in R and CatboostClassifier in Python I'm obviously missing.

I tried to look at the documentation, but I can't find anything that can help me.

In addition, I observed that when I run the predict function in R with class_type='Probability' I get a unique value per row, while running the predict function in R with class_type='Probability' too, I get 2 values per row, maybe the probability of 1 or 0, I guess. 
I don't know if this is due to the first thing I'm missunderstanding, but it would be nice if you can clarify it to me too.

I'm in Catboost 7.2 right now.
The workflow I'm following is the one given by the examples: read the dataset, create the pool objects giving the label and the categorical features to them, train, and predict.

Thank you in advance, and let me know if you need more details!

**Comment by nikitxskv** [comment]
@alvaropc Hi!
To get the same predictions try to set the same `random_seed` parameter in Python and R packages.

Python-package's predict function in case of `prediction_type='Probability'` returns the probabilities of two classes (like in sklearn package). The first column is the probabilities of the zero class and the second column is the probabilities of the first class.

R-package's predict function in case of `prediction_type='Probability'` returns only probabilities of the first class.

**Comment by alvaropc** [comment]
Hi @nikitxskv !
Thank you for the message and the explanation about the 'Probability' mode!

I still have a few problems with the predictions.
I tried to set the seed as you said in your comment, and compared the % of success while predicting over the train. Doing so, I have a huge difference in the results I obtain.

Let me copy here both codes:

R:
```
library(catboost)
train <- readRDS('/path/to/train.rds')
train_cat <- train
ID <- 'id_name'
target <- 'target_name'


train_cat[ , c(ID, target)] <- NULL
train_pool <- catboost.load_pool(train_cat,label=as.integer(train[ , target]),
                           cat_features = 1:10)
fit_params <- list(iterations = 500, random_seed=63)
cbc_pool <- catboost.train(train_pool, test_pool =  NULL, fit_params)
prediction <- catboost.predict(cbc_pool, train_pool,prediction_type = 'Class')
#Percentage of success while predicting
result <- sum((prediction==train[, target]))/length(prediction)
```
Python:

```
from catboost import Pool, CatBoostClassifier
import rpy2.robjects as robjects
from rpy2.robjects import pandas2ri
pandas2ri.activate()
readRDS = robjects.r['readRDS']
####
#Read dataset
train = readRDS('/path/to/train,rds')
#Transform from r2py to pandas
train=pandas2ri.ri2py_dataframe(train)
ID='id_name'
target='target_name'
train_cat = train.drop(columns=[ID,target])
###
train_pool=Pool(train_cat,label=train[target],cat_features=[0,1,2,3,4,5,6,7,8,9])
cbc_pool = CatBoostClassifier(iterations=500,random_seed=63).fit(train_pool)
prediction = cbc_pool.predict(train_pool, prediction_type='Class')
result=sum(prediction==train[target])/len(prediction)
```
While the result in R is 0.01525188, the reesult in python is 0.9896429716641133.
I counted the number of 1 in train target in both languages just to be sure, and in both I have 3854 ones. In predictions, I have 535 ones in Python and 363162 in R. 

I took a look just in case the zeros and ones were reversed in R, but it is not the case.

In addition, while predicting using  prediction_type='Probability', the values are quite different too, for instance, the first 3 values in Python (probability of 1) are: 0.00558235, 0.08274157 and 0.01560501, while the first 3 values in R are 0.5043396, 0.5191888 and 0.5184334. (To do so I just changed the 'Class' to 'Probability' in the code above and took a look at the first values).

I hope this is just due to something I am not seeing about my code, or due to something I am not using correctly in Catboost, but I can't find it by myself.

It would be nice to have some help! :)

Thank you so much in advance,

**Comment by nikitxskv** [comment]
1) Indices of cat features should be zero-based. So you need to replace
`train_pool <- catboost.load_pool(train_cat,label=as.integer(train[ , target]), cat_features = 1:10)`
with
`train_pool <- catboost.load_pool(train_cat,label=as.integer(train[ , target]), cat_features = 0:9)`

2) Also `CatBoostClassifier` class has `'Logloss'` as default of `loss_function` parameter. `CatBoost` class and `catboost.train` function have `'RMSE'` as default of `loss_function` parameter. So you need to replace
`fit_params <- list(iterations = 500, random_seed=63)`
with
`fit_params <- list(iterations = 500, random_seed=63, loss_function='Logloss')`

Try to do this and repeat the comparison again.

**Comment by alvaropc** [comment]
Hi again @nikitxskv !

Thank you again for the message!

1.- You got  me there, I usualy take care about this point and I totaly missed it this time! 

2.- Anyway, this point is what I was looking  for, I just didn't see it in the documentation.

However, I have to say that only changing the categorical features the result did not changed at all (I actually got a worse result, 0.01525188), it was the loss function change the one that let me got the result I was expecting, the 0.98.


Taking a closer look to the R documentation I have just found this in the about the loss_function:  'The specified value also determines the machine learning problem to solve.'
So I finaly was able to figure out what was going on in my code thanks to your help! 

(I don't know why I just thought that Catboost in R could detect that I was in a binary classification problem due to the Y label I was providing, that's why I was missunderstanding what happened)

Anyway, this final conclusion what exactly what I needed!
Thank you so much again!
